synapse/lib/syntax.py

```diff
@@ -30,6 +32,23 @@
 # this may be used to meh() potentially unquoted values
 valmeh = whites.union({'(', ')', '=', ',', '[', ']', '{', '}'})
 
+scmdre = regex.compile('^[a-z][a-z0-9\.]+$')
+univre = regex.compile(r'^\.[a-z0-9]+(:[a-z0-9]+)*$')
+propre = regex.compile(r'^[a-z][a-z0-9]+(:[a-z0-9]+)+(\.[a-z][a-z0-9]+)*$')
+formre = regex.compile(r'^[a-z][a-z0-9]+(:[a-z0-9]+)+$')
+
+def isPropName(name):
```

Comment:
I have a minor preference for using fullmatch with the ^ $ because it makes the regexes easier to compose later, but it is a minor quibble (for example you could literally `+ univre` at the end of `propre`)